pkg/config/config.go

```diff
@@ -109,16 +109,13 @@ func New(v *viper.Viper, co *Options) (*Config, error) {
 		}
 	case types.AWSSecretsManagerbackend:
 		{
-			if !v.IsSet(types.EnvAWSAccessKey) || !v.IsSet(types.EnvAWSSecretAccessKey) {
-				return nil, fmt.Errorf("Must provide %s and %s for backend type %s",
-					types.EnvAWSAccessKey,
-					types.EnvAWSSecretAccessKey,
-					types.AWSSecretsManagerbackend,
-				)
-			}
-			s := session.Must(session.NewSession(&aws.Config{
+			s, err := session.NewSession(&aws.Config{
```

Comment:
This looks fine. The only problem here is that if someone is setting their AVP config through a k8s Secret, or a config file, then specifying the `AWS_BLAH` parameters will not get sent to the AWS SDK, and it won't. Perhaps you can add some code to loop through the Viper settings and set all `AWS_` config keys as env variables like we do for `VAULT_`: https://github.com/IBM/argocd-vault-plugin/blob/main/pkg/config/config.go#L159

Reply:
Unsure of the use-cases here as I wouldn't add AWS credentials to an AVP config file/secret on k8s, preferring IAM auth or envFrom from a dedicated AWS secret instead. But others might see a use for it, so I'll add support for setting envs vars for each `AWS_` entry in the config file.

Reply:
@jkayani I just pushed e846f41 with a unit test for `AWS_` config entries. Hope I understood you right.

Reply:
Note, the fake credentials are taken from https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-set